libs/ledger-live-common/src/deviceSDK/tasks/core.ts

```diff
@@ -52,3 +74,99 @@ export function sharedLogicTaskWrapper<TaskArgumentsType, TaskEventsType>(
     });
   };
 }
+
+// To update once createCustomErrorClass is not used on Transports errors
+type ErrorClass = ReturnType<typeof createCustomErrorClass>;
+
+// To be able to retry a command, the command needs to take an object containing a transport as its argument
+type CommandTransportArgs = { transport: Transport };
+
+/**
+ * Calls a command and retries it on given errors. The transport is refreshed before each retry.
+ *
+ * The no response timeout is handled at the task level
+ *
+ * @param command the command to wrap, it should take as argument an object containing a transport
+ * @param transportRef a reference to the transport, that can be updated/refreshed
+ * @param refreshTransport an async function to refresh the transport
+ * @param allowedErrors a list of errors to retry on
+ *  - errorClass: the error class to retry on
+ *  - maxRetries: the maximum number of retries for this error, or "infinite" (TODO: don't know if we actually want an infinte option)
+ */
+export function retryOnErrorsCommandWrapper<
+  CommandArgsWithoutTransportType,
+  CommandEventsType
+>({
+  command,
+  refreshTransport,
+  transportRef,
+  allowedErrors,
+}: {
+  command: (
+    args: CommandArgsWithoutTransportType & CommandTransportArgs
```

Comment:
can't we put the `withRefreshableTransport` inside here? this way we don't need to pass the refreshTransport and transportRef arguments. This way the task doesn't even have to know about the existence of `withRefreshableTransport`.

Either way these are tightly coupled, using this wrappers **requires** the task to use withRefreshableTransport, so we might as well put it here.

Reply:
I agree that  `retryOnErrorsCommandWrapper` is coupled to `withRefreshableTransport`

But i don't see how putting the `withRefreshableTransport` inside the wrapper would work ? The idea of having the `withRefreshableTransport` is to do the "logic that was done inside withDevice" only once. And then we refresh the transport if there is an issue

By putting the `withRefreshableTransport` you are going to call it on every wrapped command no ? And you are going to call another `withDevice` for all the other commands no ? And how are you going to keep the transport up-to-date if you both use a `withDevice` and a `withRefreshableTransport` ?